Cargo.toml

```diff
@@ -40,6 +40,7 @@ wild = "2.0.4"
 owo-colors = "3.5.0"
 human_format = "1.0.3"
 once_cell = "1.19"
+globset = "0.4.14"
```

Comment:
Nvm, I don't know how I have missed it but it does look like glob provides a method that I was looking for.

Reply:
Actually, searching through 4m entries is already reasonably fast at 0.7s. In the picture that follows there are three searches, two that only had a couple of matches, and one that had 225k matches.

<img width="1440" alt="Screenshot 2023-12-19 at 15 27 11" src="https://github.com/Byron/dua-cli/assets/63622/a6d8e595-6ba8-4989-a000-067d29e08ddb">

From the numbers it's clear that the longest time is not spent displaying the 225k matches or anything, it's to get an fstat on the whole list. Fair enough, as this is a known issue - it's unable to only fstat the entries in view.

That's not to say that optimizations aren't possible, but to my mind it starts out at a good spot.

Reply:
Looks like this: https://crates.io/crates/cargo-instruments TIL!

Reply:
Actually, I wasn't using `cargo-instruments`, but attached manually. This can be a real pain, so I am happy that *I* learned something today, and will definitely use `cargo instruments` more.